### ISSUE TITLE
Add better theme options for colour blindness

### ISSUE THREAD

**Issue by vkbo** [vkbo/novelWriter#2689]
We have high contrast themes, and monochrome themes, but no dedicated themes for some types of colour blindness. It would be nice to add some.

Possible helper tool: https://rgblind.com/color-blind-palette

**Comment by vkbo** [comment]
Maybe you have time to look at this, @HeyMyian? Since you're the theme expert! 😃

**Comment by paulkater** [comment]
My colour blindness is Deuteranopia. Currently I use the old "Light theme" with black letters and clear, blue dialogue.

This as an example:

<img width="457" height="180" alt="Image" src="https://github.com/user-attachments/assets/1a68bab5-57c3-4e82-90f9-198b8c050d65" />

This one is perfect for me. #000000 for the text, as close to #0000FF as possible for the blue and I'm a happy camper.
I need colours that scream at me. 😉 
Mauve, beige, lilac and soft pink are only words for me, not colours.

**Comment by HeyMyian** [comment]
We can probably do something here.

I did some subjective tests just now with a color blindness filter (Protanopia, Deuteranopia, Tritanopia, at 100% severity respectively) and looked at how well `text`, `dialog`, and `emphasis` could be distinguished from each other.

I found the best existing themes for...
- Protanopia: Cyberpunk, Vivid White Green, Vivid Blacks, New Moon, Selenium Light
- Deuteranopia: Vivid White Green, Vivid Black Green, New Moon, Cyberpunk
- Tritanopia: Snazzy, Suburb Light, Dracula, New Moon

I used https://www.color-blind-test.com/images/simulator/ and https://github.com/DaltonLens/DaltonLens

<details>
<summary>My quick raw chaotic subjective data, higher number = better</summary>

| Dark Themes            | P | D | T |
|------------------|---|---|---|
| Warpgate         | 0 | 2 | 0 |
| Waterlily        | 0 | 1 | 0 |
| Vivid Bl Red     | 3 | 2 | 2 |
| Vivid Bl Green   | 3 | 3 | 2 |
| Tomorrow N       | 2 | 1 | 0 |
| Tomorrow N 80    | 1 | 1 | 1 |
| Tomorrow N Bri   | 1 | 1 | 0 |
| Tomorrow N Blu   | 0 | 0 | 0 |
| Tango            | 1 | 1 | 2 |
| Sultana          | 0 | 0 | 2 |
| Solarized        | 2 | 2 | 0 |
| Sepia            | 0 | 0 | 1 |
| Selenium         | 2 | 2 | 0 |
| Ruby             | 1 | 2 | 3 |
| Rosé Pine        | 1 | 2 | 0 |
| Primer           | 2 | 2 | 3 |
| Paragon          | 1 | 1 | 1 |
| Otium            | 1 | 1 | 0 |
| Nord             | 0 | 0 | 0 |
| Noctis           | 1 | 1 | 3 |
| Night Owl        | 1 | 1 | 0 |
| New Moon         | 3 | 3 | 3 |
| LCARS            | 1 | 1 | 1 |
| Jewel Case       | 1 | 1 | 0 |
| Horizon          | 1 | 1 | 2 |
| Grey Dark        | - | - | - |
| Floral Midnight  | 1 | 0 | 1 |
| Everforest       | 0 | 0 | 1 |
| Espresso         | 2 | 1 | 2 |
| Dracula          | 2 | 1 | 4 |
| Cyberpunk        | 4 | 3 | 1 |
| Chernozem        | 1 | 1 | 2 |
| Catppuccin       | 1 | 1 | 0 |
| Castle           | 1 | 1 | 0 |
| Blue Streak      | 2 | 2 | 2 |
| Modern           | 1 | 0 | 2 |
| Classic          | 0 | 0 | 1 |
| Suburb           | 0 | 1 | 3 |
| Garden           | 2 | 2 | 2 |
| Aura             | 1 | 0 | 1 |
| Aura Soft        | 0 | 0 | 0 |
| Default          | 2 | 2 | 3 |

| Bright Themes            | P | D | T |
|------------------|---|---|---|
| Default          | 2 | 2 | 3 |
| Aura Bright      | 1 | 0 | 0 |
| Garden           | 1 | 1 | 2 |
| Suburb           | 0 | 0 | 4 |
| Classic          | 1 | 1 | 1 |
| Modern           | 0 | 0 | 2 |
| Blue Streak      | 1 | 1 | 1 |
| Castle           | 1 | 1 | 2 |
| Catppuccin       | 1 | 1 | 0 |
| Chalky Soil      | 1 | 0 | 2 |
| Everforest       | 3 | 3 | 1 |
| Floral Daydream  | 0 | 1 | 1 |
| Full Moon        | 2 | 2 | 2 |
| Grey             | - | - | - |
| Horizon          | 1 | 2 | 2 |
| Jewel Case       | 1 | 1 | 1 |
| Light Owl        | 2 | 1 | 3 |
| Noctis Lux       | 1 | 1 | 2 |
| Nordlicht        | 0 | 0 | 1 |
| Otium            | 1 | 1 | 2 |
| Primer           | 2 | 1 | 1 |
| Rosé Pine Dawn   | 0 | 0 | 0 |
| Ruby             | 0 | 1 | 3 |
| Selenium         | 3 | 3 | 0 |
| Sepia            | 0 | 1 | 2 |
| Snazzy           | 1 | 1 | 4 |
| Solarized        | 2 | 2 | 1 |
| Sultana          | 0 | 0 | 3 |
| Tango            | 2 | 1 | 3 |
| Tomorrow         | 1 | 1 | 0 |
| Vivid White Gr   | 4 | 4 | 1 |
| Vivid White Red  | 0 | 1 | 2 |
| Waterlily        | 0 | 0 | 2 |
</details>

But I didn't look at all at the other aspects, like the `replacetag` in the screenshot above.

What I found works well is to work with brightness too; e.g. the themes that have grey-ish text and `#ffffff` dialogue have one "free" distinction built in that might work for all users.

I think it's possible to create a theme that is better for different variations of colorblindness than the ones I mentioned above. I suspect though that some syntax highlighting will have to fall by the wayside. Which ones are most important will vary depending on how novelWriter is used individually... 🤔 Some users need the replacetags, others need %Note, some might not care for emphasis...?

**Comment by vkbo** [comment]
Maybe we can make a theme for each variation, and limit the syntax highlighting to 4 or 5 colours? It's fine to use the same colour for multiple things as long as we can avoid the same colour stacking on top of another. It's possible to extend the range with shades. Note/story and key can be two shades of the same colour.

I can give this a try as well, and we can see if we can converge at a common way of making three variations of a theme.

**Comment by paulkater** [comment]
Pitching in as a colour blind person: having limited colours in a theme should not be a problem. If e.g. the colours for %synopsis and %comment are the same as the `<replacetag>` that is fine, they're usually not used on top of each other.

Please remember: many colour blind people don't see many colours and shades. Instead of marine blue, navy blue, royal blue and whatever blue I see blue. Or perhaps dark grey. Limiting a high contrast theme to 3 colours (white or black background, 1 for %-options and replacetag and 1 for dialogue) and black (white theme) / white (dark theme) would already be a big improvement.

> Deuteranopia: Vivid White Green, Vivid Black Green, New Moon, Cyberpunk

I tried all of them. Nice work for sure but the difference in colours is just too little for me. The non-white background isn't helping, and the vague 'greyish' dialogue colours on that background also make this difficult for me. Even just reading it, in fact. I do understand that this might be a serious issue to decide on as there are many versions of colour blindness.

Drawing from professional experience:

The way a supplier of support ticket software fixed this for me in their coding section (where errors were shown in red and I didn't see them) was to simply underline those errors with tiny dots, a bit like the squiggly lines that a spell checker uses. Maybe that is something to look into? Don't focus on colours on everything for the colour blind but put the replace tags in bold, like < **replace me** >, simply in black/white? That would save a colour and might make things easier.

Then all we'd need could be:

Light: background white, %-options dark grey, dialogue bright blue, text black.
Dark: background dark grey/black, %-options middle grey, dialogue yellow or bright green, text white.

Again, and I do want to repeat this because it's not something colourful people realise: colour blindness means what it says. The inability to see many different colours. I don't know what beige looks like, or purple, etc. I can call something yellow when it's "neon green". Or green when it's light brown.

Sorry for being so wordy about this but I'm trying to make you see my colours. ;-)

**Comment by vkbo** [comment]
I appreciate the input. It helps a lot.

What I meant by shades in this context was just making it slightly darker to indicate a variation, but not needing to make distinct colours. I generally try to avoid shades – as in tone variations – of the same colour. That is usually a bad thing to do with themes, aside from shades of white/grey. novelWriter uses 7 base colours for the themes, and my intention here is to try to merge them down to, say, 5 that are easy to distinguish and build a set of 3 light and 3 dark themes from that.

**Comment by vkbo** [comment]
About your suggestion for other ways of indicating difference: There is already an option to just show emphasis rather than or in addition to use colour. This can be applied to more features I think. It is trivial to fix it in the syntax highlighter as it just applies pre-defined formats.

How about we add an Accessibility section below the theme section, and add some more options like that. For instance, replace tags and shortcodes with **bold**. I can add a dropdown of options with, say bold or underline with and without colour and just plain colour as the default. I can move the colouring of emphasis to the same section. I can also add a bold flag for synopsis/note/story comments with and without colour.

With this, we can reserve the clear colour differences for dialogue and notes.

Does that sound like a solution?

**Comment by paulkater** [comment]
Wow! Yes! The accessibility idea really sounds like a great way to go. It might also help people who are otherwise visually impaired!

**Comment by vkbo** [comment]
I added a separate task for the preferences settings: #2691

**Comment by paulkater** [comment]
I just had an idea which may be too simple:
If we take these settings:

<img width="518" height="249" alt="Image" src="https://github.com/user-attachments/assets/073657e8-9b63-4cc6-86bf-3528a2044f41" />

would it be an option to add 3 straight-forward optional input fields to override a theme, e.g.

Background:
Font colour:
Dialogue colour:
where someone can enter a colour name (with support from a limited dropdown even?)

e.g.
Background: white
Font colour: black
Dialogue colour: blue

or the #RGB code for the colour:
Background: #FFFFFF
Font colour: #000000
Dialogue colour: #0000FF

and (ideally) a small box behind each option showing the colour.

Or is that too much interference with the theme engine?

**Comment by vkbo** [comment]
I have though about adding a tool where you can copy a theme and make your own edits. But here I prefer to just add two new themes for red/green and blue colour blindness.

Is there a second colour code that is easy to distinguish for you? One with any combination of the `#XXXX00` in `#RGB` syntax?

**Comment by vkbo** [comment]
How are these colours contrasted against `#0000FF`?

<img width="220" height="116" alt="Image" src="https://github.com/user-attachments/assets/9c9e2665-02b6-45f2-9b8a-14aa3943a973" />

**Comment by paulkater** [comment]
The third one, #757500, is fabulous! I have no idea what it's called but that's great. <3

**Comment by vkbo** [comment]
How does this look to you?

<img width="2325" height="1347" alt="Image" src="https://github.com/user-attachments/assets/345a4ff5-f0c3-45f2-98b9-8f124feaf9e5" />

**Comment by paulkater** [comment]
This is fabulous. I love it.

**Comment by vkbo** [comment]
> The third one, #757500, is fabulous! I have no idea what it's called but that's great. <3

It's the simulated deuteranopia look of bright red actually. The second is green, and the first is orange I believe. My example theme above uses a mix of blue, green and yellow. I suspect the green and yellow looks like two shades of the same to you. I've avoided red altogether.

> This is fabulous. I love it.

Excellent. I've created the theme so I can alternate between full colour and simulated deuteronopia just to check that the contrasts make sense still. I'm glad it does.

**Comment by paulkater** [comment]
Yes, the first two seem to be family, one a paler version and the middle is the 'tanned' one.
I appreciate your patience with this so much!

**Comment by vkbo** [comment]
Ok, I've created two themes. The Deuteranopia one is tweaked a little from the above. Mainly just darker text.

I will merge this into the beta branch for 26.1, but if you wish to test it with 2.8.x, you can just grab the theme file and copy it into the `novelwriter/assets/themes` folder in your installation locally. It is compatible. I plan to make the beta release this Easter though, so it shouldn't be long before a test release is available. That way we can tweak them for the full release.

Here are the new themes:

**Today Deuteranopia Light**

<img width="2325" height="1347" alt="Image" src="https://github.com/user-attachments/assets/94c867ad-cbd5-42b2-aed8-d514c1da29a8" />

**Today Tritanopia Light**

<img width="2325" height="1347" alt="Image" src="https://github.com/user-attachments/assets/119bc34b-1659-4c54-9608-d1229108d678" />

**Comment by paulkater** [comment]
Thank you so much; I'm running the beta and this is wonderful.
